### ISSUE TITLE
Validate length/illegal characters in task descriptions

### ISSUE THREAD

**Issue by 4oo4** [nextcloud/tasks#79]
### Steps to reproduce
1. Create a task with a very long description (see example below)
2. Reload Tasks app, try to save new tasks

### Expected behaviour
The task gets saved and the tasks in the list appear on page load.

### Actual behaviour
When the length of one task's description exceeds a certain length (or it contains certain data), it bleeds into other tasks or calendars and causes an empty task list after reloading the page. After deleting the one task with the long description, everything worked normally. The only errors I found were in the browser console (see below). This was an error on my part, but after tearing everything apart to find out the cause, I thought it would be worth mentioning to see if some kind of validation can be done on the input to prevent it from happening.

The task description was as follows (this was hastily pasted from a web page, so I wasn't sure if the length of the task or something else caused the XML to be malformed):
```
Analyze this line[31m[!][0m Title: WordPress 2.3-4.7.5 - Host Header Injection in Password Reset
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8807
Analyze this lineReference: https://exploitbox.io/vuln/WordPress-Exploit-4-7-Unauth-Password-Reset-0day-CVE-2017-8295.html
Analyze this lineReference: http://blog.dewhurstsecurity.com/2017/05/04/exploitbox-wordpress-security-advisories.html
Analyze this lineReference: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-8295
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 2.7.0-4.7.4 - Insufficient Redirect Validation
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8815
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/76d77e927bb4d0f87c7262a50e28d84e01fd2b11
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this line[34m[i][0m Fixed in: 4.7.5
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 2.5.0-4.7.4 - Post Meta Data Values Improper Handling in XML-RPC
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8816
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/3d95e3ae816f4d7c638f40d3e936a4be19724381
Analyze this line[34m[i][0m Fixed in: 4.7.5
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 3.4.0-4.7.4 - XML-RPC Post Meta Data Lack of Capability Checks
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8817
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/e88a48a066ab2200ce3091b131d43e2fab2460a4
Analyze this line[34m[i][0m Fixed in: 4.7.5
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 2.5.0-4.7.4 - Filesystem Credentials Dialog CSRF
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8818
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/38347d7c580be4cdd8476e4bbc653d5c79ed9b67
Analyze this line[34m[i][0m Fixed in: 4.7.5
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 3.3-4.7.4 - Large File Upload Error XSS
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8819
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/8c7ea71edbbffca5d9766b7bea7c7f3722ffafa6
Analyze this line[34m[i][0m Fixed in: 4.7.5
Analyze this line
Analyze this line[31m[!][0m Title: WordPress 3.4.0-4.7.4 - Customizer XSS & CSRF
Analyze this lineReference: https://wpvulndb.com/vulnerabilities/8820
Analyze this lineReference: https://wordpress.org/news/2017/05/wordpress-4-7-5/
Analyze this lineReference: https://github.com/WordPress/WordPress/commit/3d10fef22d788f29aed745b0f5ff6f6baea69af3
Analyze this line[34m[i][0m Fixed in: 4.7.5
```

### Server configuration
**Operating system**:
Ubuntu 16.04
**Web server:**
nginx
**Database:**
mysql
**PHP version:**
7.0
**Nextcloud version:** (see Nextcloud admin page)
11.0.3
**Tasks version:** (see Nextcloud apps page)
0.9.5
**Updated from an older Nextcloud or fresh install:**
Updated

**Signing status:**
Pass

**List of activated apps:**

```
Enabled:
 - activity: 2.4.1
 - admin_audit: 1.1.0
 - bookmarks: 0.9.1
 - calendar: 1.5.2
 - comments: 1.1.0
 - contacts: 1.5.3
 - dav: 1.1.1
 - federatedfilesharing: 1.1.1
 - federation: 1.1.1
 - files: 1.6.1
 - files_antivirus: 1.0.0.0
 - files_opds: 0.8.2
 - files_pdfviewer: 1.0.1
 - files_reader: 1.0.4
 - files_sharing: 1.1.1
 - files_texteditor: 2.2
 - files_trashbin: 1.1.0
 - files_versions: 1.4.0
 - files_videoplayer: 1.0.0
 - firstrunwizard: 2.0
 - gallery: 16.0.0
 - keeweb: 0.3.1
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.0
 - notifications: 1.0.1
 - ownnote: 1.08
 - password_policy: 1.1.0
 - provisioning_api: 1.1.0
 - serverinfo: 1.1.1
 - sharebymail: 1.0.1
 - survey_client: 0.1.5
 - systemtags: 1.1.3
 - tasks: 0.9.5
 - theming: 1.1.1
 - twofactor_backupcodes: 1.0.0
 - updatenotification: 1.1.1
 - workflowengine: 1.1.1

Disabled:
 - bruteforcesettings
 - documents
 - encryption
 - external
 - files_accesscontrol
 - files_automatedtagging
 - files_external
 - files_retention
 - richdocuments
 - spreedme
 - templateeditor
 - user_external
 - user_ldap
 - user_saml
 - user_ldap
 - user_saml
```

**Nextcloud configuration:**

```
{
   "system": {
       "instanceid": "***REMOVED SENSITIVE VALUE***",
       "passwordsalt": "***REMOVED SENSITIVE VALUE***",
       "secret": "***REMOVED SENSITIVE VALUE***",
       "trusted_domains": [
           "***REMOVED SENSITIVE VALUE***",
           "***REMOVED SENSITIVE VALUE***"
       ],
       "datadirectory": "\/mnt\/ext\/.nextcloud\/data",
       "overwrite.cli.url": "https:\/\/example.com\/nextcloud",
       "htaccess.RewriteBase": "\/nextcloud",
       "dbtype": "mysql",
       "version": "11.0.3.2",
       "dbname": "***REMOVED SENSITIVE VALUE***",
       "dbhost": "localhost",
       "dbtableprefix": "oc_",
       "dbuser": "***REMOVED SENSITIVE VALUE***",
       "dbpassword": "***REMOVED SENSITIVE VALUE***",
       "logtimezone": "UTC",
       "installed": true,
       "theme": "",
       "maintenance": false,
       "loglevel": 0,
       "appstore.experimental.enabled": true,
       "updater.release.channel": "stable",
       "updater.secret": "***REMOVED SENSITIVE VALUE***",
       "defaultapp": "tasks",
       "trusted_proxies": [
           "127.0.0.1"
       ],
       "forwarded_for_headers": [
           "HTTP_X_FORWARDED_FOR"
       ],
       "auth.bruteforce.protection.enabled": true,
       "enabledPreviewProviders": [
           "OC\\Preview\\Image",
           "OC\\Preview\\MP3",
           "OC\\Preview\\TXT",
           "OC\\Preview\\MarkDown",
           "OC\\Preview\\Epub",
           "OC\\Preview\\PDF",
           "OC\\Preview\\OpenDocument",
           "OC\\Preview\\StarOffice",
           "OC\\Preview\\MSOfficeDoc",
           "OC\\Preview\\MSOffice2003",
           "OC\\Preview\\MSOffice2007",
           "OC\\Preview\\FB2"
       ],
       "mail_smtpmode": "php",
       "memcache.local": "OC\\Memcache\\Redis",
       "memcache.locking": "OC\\Memcache\\Redis",
       "redis": {
           "host": "localhost",
           "port": 6379,
           "dbindex": 0,
           "timeout": 1.5
       },
       "mail_smtpsecure": "tls",
       "mail_smtpauthtype": "LOGIN",
       "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
       "mail_smtpport": "587",
       "mail_smtpauth": 1,
       "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
       "mail_smtppassword": "***REMOVED SENSITIVE VALUE***"
   }

}
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Browser:**

**Operating system:**
Windows 7
**CalDAV-clients:**
N/A

#### Web server error log
N/A

#### Nextcloud log (data/nextcloud.log)
N/A

#### Browser log
```
XML Parsing Error: not well formed location: https://example.com/nextcloud/remote.php/dav/calendars/user/calendar1
XML Parsing Error: not well formed location: https://example.com/nextcloud/remote.php/dav/calendars/user/calendar2
```

**Comment by raimund-schluessler** [comment]
This sounds like a bug in Nextcloud server. Could you please check if this still occurs in Nextcloud 12.0.1?

**Comment by 4oo4** [comment]
Forgot about this until now, it appears to be fixed with 12.0.2. After upgrading I'm not able to reproduce the issue. Thanks for the tip!
